src/index.js

```diff
@@ -0,0 +1,36 @@
+const core = require("@actions/core");
+const childProcess = require("child_process");
+const { saucectlInstall } = require("./install");
+const { saucectlRun } = require("./run");
+const { awaitExecution } = require("./helpers");
+
+const config = require("./config");
+
+async function run() {
+    const cfg = config.get();
+    if (!cfg) {
+        core.setFailed("Invalid configuration.");
+        return;
+    }
+
+    // Install saucectl
+    if (!await saucectlInstall({ versionSpec: cfg.saucectlVersion })) {
+        return;
+    }
+
+    // Run it to confirm version
+    const child = childProcess.spawn('saucectl', ['--version']);
+    child.stdout.pipe(process.stdout);
+    child.stderr.pipe(process.stderr);
+    const exitCode = await awaitExecution(child);
+    core.info(`ExitCode: ${exitCode}`);
+
+    // Really execute saucectl
+    if (!cfg.skipRun) {
+        if (!await saucectlRun(cfg)) {
+            return;
+        }
+    }
+}
+
+run();
```

Comment:
This will allow to import and also test the run method because we will only call it if executed directly.

```suggestion
if (require.main === module) {
    run();
}
```